src/map.ts

```diff
@@ -28,7 +28,7 @@ const nonBlockingLoader = (loader: RequireLoader | ImportLoader) => new Promise(
 
 export const getComponent = async (name: string) => {
     if (!isCached(name)) {
-        const { require: load, loader, ...rest } = mapLoadable[name];
+        const { require: load, loader, ...rest } = mapLoadable.get(name)!;
```

Comment:
Well, I think this code is also dangerous. I mean I understand why it's written in such way, but if user hasn't registered a component and will try to render it, then we will show an exception in style like "can not read property from undefined", and it doesn't really help a lot...

Maybe we can add code like:

```ts
const loadable = mapLoadable.get(name);

if (!loadable) {
  console.error(`You are trying to render a component "${name}" that hasn't been registered yet! Please, be sure that you register a component before attempting to render it.`);
  return null;
}
```

In this case TS will understand that it's alright and user will have a better understanding what triggered this exception. What do you think?

Reply:
So, I think it may happen in your previous implementation too and it can be done in another pull request 🙂